com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/SetBreakpointsRequestHandler.java

```diff
@@ -130,6 +146,55 @@ public void initialize(IDebugAdapterContext context) {
         }
     }
 
+    private void registerBreakpointHandler(IDebugAdapterContext context) {
+        IDebugSession debugSession = context.getDebugSession();
+        if (debugSession != null) {
+            debugSession.getEventHub().events().subscribe(debugEvent -> {
+                if (!(debugEvent.event instanceof BreakpointEvent)) {
```

Comment:
use observable.filter to only process BreakpointEvent.

Reply:
Done in PR2